### PR TITLE
Fix Files.app extension

### PR DESCRIPTION
**Pull request by adrien-coye**
- Fix Realm using more than 20MiB in Files.app.
- Tune upload mechanism to use ~4MiB max during uploads in Files.app.
- Fix async issues resulting in inconsistencies in Files.app.
- Disabled some notifications when uploading from Files.app.

### CODE REVIEW

File: kDriveFileProvider/FileProviderExtension+Actions.swift [hunk 1]
```diff
@@ -20,43 +20,60 @@ import FileProvider
...
 
-        Task {
             do {
-                let response = try await driveFileManager.apiFetcher.deleteDefinitely(file: ProxyFile(driveId: driveFileManager.drive.id, id: fileId))
+                let response = try await self.driveFileManager.apiFetcher
+                    .deleteDefinitely(file: ProxyFile(driveId: self.driveFileManager.drive.id, id: fileId))
                 if response {
-                    FileProviderExtensionState.shared.workingSet.removeValue(forKey: itemIdentifier)
+                    self.fileProviderState.removeWorkingDocument(forKey: itemIdentifier)
                     self.manager.signalEnumerator(for: .workingSet) { _ in }
```
Comment (adrien-coye) [inline kDriveFileProvider/FileProviderExtension+Actions.swift:77]: switch to async/await

File: kDriveCore/Data/Cache/DriveFileManager.swift [hunk 2]
```diff
@@ -69,6 +83,11 @@ public class DriveFileManager {
                 if oldSchemaVersion < 13 {
                     migration.deleteData(forType: UploadFile.className())
                 }
+
+                // Migration for Upload APIV2 build3
+                if oldSchemaVersion < 14 {
```
Comment (PhilippeWeidmann) [inline kDriveCore/Data/Cache/DriveFileManager.swift:88]: Since we are doing the same for 13 and 14 we can simply remove the block for 13

Comment (adrien-coye) [reply kDriveCore/Data/Cache/DriveFileManager.swift:83]: nice catch

File: kDriveCore/FileProvider/FileProviderItem.swift [hunk 3]
```diff
@@ -19,9 +19,8 @@
 import CoreServices
 import FileProvider
 import InfomaniakDI
-import kDriveCore
 
-extension NSFileProviderItemIdentifier {
+public extension NSFileProviderItemIdentifier {
```
Comment (PhilippeWeidmann) [inline kDriveCore/FileProvider/FileProviderItem.swift:23]: Why do we need it to be public ?

**Pull request merged**